src/main.ts

```diff
@@ -141,69 +126,160 @@ async function getWithRetry(client: AxiosInstance, url: string, retryCondition:
   throw new Error('Max retries reached');
 }
 
+async function getStickyDisk(dockerfilePath: string, retryCondition: (error: AxiosError) => boolean): Promise<any> {
```

Comment:
Instead of doing the `Promise.race` in the caller, its more canonical to use an [`AbortController` ](https://developer.mozilla.org/en-US/docs/Web/API/AbortController)